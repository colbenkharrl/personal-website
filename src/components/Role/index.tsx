import Image, { ImageProps } from 'next/image';
import Link from 'next/link';

export interface Role {
  company: string;
  title: string;
  logo: ImageProps['src'];
  start: string | { label: string; dateTime: string };
  end: string | { label: string; dateTime: string };
  url: string;
  duration: string;
}

export function Role({ role }: { role: Role }) {
  const startLabel =
    typeof role.start === 'string' ? role.start : role.start.label;
  const startDate =
    typeof role.start === 'string' ? role.start : role.start.dateTime;

  const endLabel = typeof role.end === 'string' ? role.end : role.end.label;
  const endDate = typeof role.end === 'string' ? role.end : role.end.dateTime;

  return (
    <li
      className="group relative flex items-start gap-2 sm:gap-4"
      data-testid="Role"
    >
      <div className="relative mt-1 flex h-10 w-10 flex-none items-center justify-center rounded-full shadow-md shadow-zinc-800/5 ring-1 ring-zinc-900/5 dark:border dark:border-zinc-700/50 dark:bg-zinc-800 dark:ring-0">
        <Image
          src={role.logo}
          alt=""
          className="relative z-10 h-7 w-7"
          unoptimized
        />
      </div>
      <dl className="flex flex-auto flex-wrap gap-x-2">
        <dt className="sr-only">Company</dt>
        <dd className="relative z-10 flex w-full text-sm font-medium text-zinc-900 dark:text-zinc-100">
          {role.company}
          <div className="flex-grow" />
          <div className="flex flex-col items-end">
            <p className="relative z-10 text-xs text-zinc-400 dark:text-zinc-500">
              <time className="relative z-10" dateTime={startDate}>
                {startLabel}
              </time>{' '}
              <span className="relative z-10" aria-hidden="true">
                â€”
              </span>{' '}
              <time className="relative z-10" dateTime={endDate}>
                {endLabel}
              </time>{' '}
              ({role.duration})
            </p>
          </div>
        </dd>
        <dt className="sr-only">Role</dt>
        <dd className="text-xs text-zinc-500 dark:text-zinc-400">
          <div className="absolute -inset-x-2 -inset-y-2 z-0 scale-95 rounded-lg bg-zinc-50 opacity-0 transition group-hover:scale-100 group-hover:opacity-100 dark:bg-zinc-800/50" />
          <Link href={role.url}>
            <span className="absolute -inset-x-4 -inset-y-6 z-20 sm:-inset-x-6 sm:rounded-2xl" />
            <span className="relative z-10  group-hover:text-amber-500">
              {role.title}
            </span>
            <br />
          </Link>
        </dd>
      </dl>
    </li>
  );
}
