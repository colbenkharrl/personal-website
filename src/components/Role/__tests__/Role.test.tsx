import { render, screen } from '@testing-library/react';
import * as stories from '../__stories__/Role.stories';
import { composeStories } from '@storybook/react';

const { Basic } = composeStories(stories);

it('Role matches snapshot (regression test)', () => {
  render(<Basic />);
  expect(screen.getByTestId('Role')).toMatchSnapshot(
    'Full component snapshot.',
  );
});
