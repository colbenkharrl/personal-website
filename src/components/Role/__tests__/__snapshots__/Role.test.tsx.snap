// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Role matches snapshot (regression test): Full component snapshot. 1`] = `
<li
  class="group relative flex items-start gap-2 sm:gap-4"
  data-testid="Role"
>
  <div
    class="relative mt-1 flex h-10 w-10 flex-none items-center justify-center rounded-full shadow-md shadow-zinc-800/5 ring-1 ring-zinc-900/5 dark:border dark:border-zinc-700/50 dark:bg-zinc-800 dark:ring-0"
  >
    <img
      alt=""
      class="relative z-10 h-7 w-7"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/img.jpg"
      style="color: transparent;"
      width="40"
    />
  </div>
  <dl
    class="flex flex-auto flex-wrap gap-x-2"
  >
    <dt
      class="sr-only"
    >
      Company
    </dt>
    <dd
      class="relative z-10 w-full flex-none text-sm font-medium text-zinc-900 dark:text-zinc-100"
    >
      Spec
    </dd>
    <dt
      class="sr-only"
    >
      Role
    </dt>
    <dd
      class="text-xs text-zinc-500 dark:text-zinc-400"
    >
      <div
        class="absolute -inset-x-2 -inset-y-2 z-0 scale-95 rounded-lg bg-zinc-50 opacity-0 transition group-hover:scale-100 group-hover:opacity-100 dark:bg-zinc-800/50"
      />
      <a
        href="https://specprotected.com/"
      >
        <span
          class="absolute -inset-x-4 -inset-y-6 z-20 sm:-inset-x-6 sm:rounded-2xl"
        />
        <span
          class="relative z-10  group-hover:text-orange-500"
        >
          VP of Engineering
        </span>
      </a>
    </dd>
    <dt
      class="sr-only"
    >
      Date
    </dt>
    <dd
      aria-label="2020 until 2023"
      class="relative z-10 ml-auto text-xs text-zinc-400 dark:text-zinc-500"
    >
      <time
        datetime="2020"
      >
        2020
      </time>
       
      <span
        aria-hidden="true"
      >
        â€”
      </span>
       
      <time
        datetime="2023"
      >
        2023
      </time>
    </dd>
  </dl>
</li>
`;
